doc/man7/provider.pod

```diff
@@ -260,128 +192,6 @@ MODE is only present where applicable.
 Other aliases may exist for example where standards bodies or common practice
 use alternative names or names that OpenSSL has used historically.
 
-=head1 OPENSSL PROVIDERS
```

Comment:
I'm ambivalent about moving this section to `crypto.pod`, because while some of them are built into libcrypto, they are backends, not part of the *framework* that libcrypto is becoming.  Personally, I would leave this section here.

Reply:
Linking back to crypto.pod would also be sufficient..

Reply:
Linking here from crypto.pod can be argued too

Reply:
I am firmly of the view that the text belongs in crypto.pod. I've added a link to it from here.

Reply:
After a talk with @mattcaswell, I see reason with this move.